core/widgets/SubmitButtonWidget.php

```diff
@@ -42,9 +42,31 @@ class SubmitButtonWidget extends Widget
     public $options = [];
 
     /**
+     * @since 1.0.24 First time this was introduced
+     *
      * @var activeForm Define activeForm context to use widget in context and only disable button when validation succeeded
      */
-    public $activeForm;
+    protected $activeForm;
+
+    /**
+     * @since 1.0.24 First time this was introduced
+     *
+     * @param $activeForm Set $activeForm and check for type "ActiveForm"
+     */
+    public function setActiveForm($activeForm) {
```

Comment:
you can use the type check of php:

setActiveForm(ActiveForm $activeForm)
{
   $this->_activeForm = $activeForm;
}

Reply:
regarding php doc (https://github.com/luyadev/luya/blob/master/docs/guide/luya-guideline.md#phpdoc)

```php
/**
 * Setter mehod for activeForm
 * 
 * @param ..
 * @since
 */

Reply:
I can switch it. I was referring to https://docs.phpdoc.org/references/phpdoc/tags/since.html